lnst/Tests/LongLivedConnections.py

```diff
@@ -0,0 +1,174 @@
+        self._listening_thread.start()
+
+        self._polling_thread = threading.Thread(target=self._poll_connections)
+        self._polling_thread.start()
+
+    def _stop(self):
+        self._result = (
+            True if len(self._connections) == self.params.connections_count else False
+        )
+
+        logging.info("Stopping LongLivedServer server")
+        self._running = False
+
+        self._listening_thread.join()
+        self._polling_thread.join()
+
+        for conn in self._connections:
+            conn.close()
+
+    def _poll_connections(self):
+        while self._running:  # no need to lock due to GIL
+            if not self._connections:
+                time.sleep(1)
+                continue
+
+            try:
+                ready, _, _ = select.select(self._connections, [], [], 1)
+            except ValueError:
+                continue
+
+        self._connections = []
+
+    def _listen(self):
+        with socket.socket(
+            get_inet(self.params.server_ip), socket.SOCK_STREAM
+        ) as server_socket:
+            server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            server_socket.bind((str(self.params.server_ip), self.params.server_port))
+            server_socket.listen(65536)
+
+            logging.info(
+                f"TCP server started on {self.params.server_ip}:{self.params.server_port}"
+            )
+            while self._running:
+                try:
+                    ready, _, _ = select.select([server_socket], [], [], 1)
+                except ValueError:
+                    continue
+
+                if not ready:
+                    time.sleep(1)  # prevent active polling
+                    continue
+
+                for conn in ready:
+                    client_socket, client_address = server_socket.accept()
+                    client_socket.setblocking(False)
+
+                    self._connections.append(client_socket)
+
+
+class LongLivedClient(LongLivedMixin):
+    client_ip = StrParam(mandatory=True)
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._connections = []
+
+    def _start_connection(self):
+        sck = socket.socket(get_inet(self.params.server_ip), socket.SOCK_STREAM)
+        sck.bind(
+            (self.params.client_ip, 0)
+        )  # needs to be binded to specific IP to respect flow IPs
+        sck.connect((str(self.params.server_ip), self.params.server_port))
+
+        sck.setblocking(False)
+        #  ^^^ sets socket into a "something may be written to there" state,
+        # so it won't close connection when jumping out of this function context
+
+        return sck
+
+    def _start(self):
+        for _ in range(self.params.connections_count):
+            conn = self._start_connection()
+            self._connections.append(conn)
+
+        logging.info("All connections established, waiting")
+        print("Connected to all")
```

Comment:
Debug statement leftover? Move to logging if useful, but seems the previous statement basically does the same.

Reply:
Yep, thanks.